### PR TITLE
Refactor "Relations"

### PR DESCRIPTION
**Pull request by olovy**

### CODE REVIEW

File: whelk-core/src/main/groovy/whelk/component/DependencyCache.groovy [hunk 1]
```diff
@@ -26,7 +26,7 @@ class DependencyCache {
             .refreshAfterWrite(5, TimeUnit.MINUTES)
             .build(loader(storage.&getDependersOfType))
 
-    private LoadingCache<Tuple2<String,String>, Set<String>> dependencyCache = CacheBuilder.newBuilder()
+    private LoadingCache<Tuple2<String,String>, Set<String>> dependenciesCache = CacheBuilder.newBuilder()
             .maximumSize(1000)
```
Comment (niklasl) [inline whelk-core/src/main/groovy/whelk/component/DependencyCache.groovy]: Factor out to constant? I wonder if 1000 is enough?

File: whelk-core/src/main/groovy/whelk/component/DependencyCache.groovy [hunk 2]
```diff
@@ -49,7 +49,7 @@ class DependencyCache {
             String relation = it[0]
             String iri = it[1]
             dependersCache.invalidate(new Tuple2(iri, relation))
-            dependencyCache.invalidate(new Tuple2(docIri, relation))
+            dependenciesCache.invalidate(new Tuple2(docIri, relation))
```
Comment (niklasl) [inline whelk-core/src/main/groovy/whelk/component/DependencyCache.groovy]: Could we have a key collision here with subject+relation vs. object+relation? (E.g. for `<X> :broader ?o` would we overwrite the cache for `?s :broader <Y>`?)

Comment (olovy) [reply whelk-core/src/main/groovy/whelk/component/DependencyCache.groovy]: `dependersCache` is `object+relation -> [subject1, ..., subjectN]`
`dependencyCache` is `subject+relation -> [object1, ..., objectN]`

Comment (olovy) [reply whelk-core/src/main/groovy/whelk/component/DependencyCache.groovy]: When a document is updated only its outgoing links might have changed so we clear out all cached relations where it is the subject.

Comment (niklasl) [reply whelk-core/src/main/groovy/whelk/component/DependencyCache.groovy]: Now I see it! All good then!

File: whelk-core/src/main/groovy/whelk/Whelk.groovy [hunk 3]
```diff
@@ -392,12 +392,8 @@ class Whelk implements Storage {
...
     }
 
-    boolean isImpliedBy(String broaderIri, String narrowerIri) {
-        return relations.isImpliedBy(broaderIri, narrowerIri)
-    }
-
-    Set<String> findInverseBroaderRelations(String iri) {
-        return relations.findInverseBroaderRelations(iri)
+    Relations relations() {
+        return relations
```
Comment (niklasl) [inline whelk-core/src/main/groovy/whelk/Whelk.groovy]: getRelations() ?

Comment (olovy) [reply whelk-core/src/main/groovy/whelk/Whelk.groovy]: For me `relations()` reads better, but I changed it to be consistent.

Comment (niklasl) [reply whelk-core/src/main/groovy/whelk/Whelk.groovy]: I see your POV, I just think in "bean property" terms here (and will use it as `whelk.relations` in groovy...). We don't have much of a "fluent" style, so yes, consistent too.

**Pull request merged**